### DISCUSSION TITLE
Allow passing public RPC endpoint to the CLI

### DISCUSSION THREAD

**Discussion by 0237h** [graphprotocol/graph-tooling#1758] [Ideas]
In `graph init` for example, if a start block lookup fails due to a faulty RPC endpoint, the user has to then input it manually.

It would be useful to have a CLI option that allows specifying a different public RPC endpoint from the [hard-coded default](https://github.com/graphprotocol/graph-tooling/blob/d764867dc7eab32e2f1db8505419ea4fa03ed503/packages/cli/src/command-helpers/abi.ts#L347).

Something like:
```
graph init --public-rpc <endpoint>
```

**Comment by YaroShkvorets** [comment]
With the upcoming [registry](https://github.com/graphprotocol/networks-registry) release `graph init` will try all available RPC endpoints. Fault RPC endpoints shouldn't exist, and if that happens we should encourage making PRs to the registry instead.
